cpp/dolfinx/common/Timer.h

```diff
@@ -28,36 +28,40 @@ namespace dolfinx::common
 ///
 ///   list_timings();
 
+template <typename chrono_timer = std::chrono::high_resolution_clock>
 class Timer
 {
 public:
   /// Create timer
   ///
   /// If a task name is provided this enables logging to logger, otherwise (i.e.
   /// no task provided) nothing gets logged.
-  Timer(std::optional<std::string> task = std::nullopt);
+  Timer(std::optional<std::string> task = std::nullopt) : _task(task) {}
 
   /// Destructor
-  ~Timer();
+  ~Timer() = default;
 
   /// Zero and start timer
-  void start();
-
-  /// Resume timer. Not well-defined for logging timer
-  void resume();
+  void start() { _start_time = chrono_timer::now(); }
 
   /// Stop timer, return wall time elapsed and store timing data into
   /// logger
-  double stop();
-
-  /// Return wall, user and system time in seconds
-  std::array<double, 3> elapsed() const;
+  template <typename unit = std::chrono::nanoseconds>
+  double stop()
```

Comment:
It would be neat to return `std::chrono::duration`. The user can then handle it. `nanobind` can turn it into a Python object.